### PR TITLE
Promotion banner

### PR DESCRIPTION
**Pull request by mic1000**
This is the new promotion banner component. Today, the "marketing image banner" is only used on the accounts overview page. This promotion banner can appear on any page with either a circle image or Icon Badge, along with a title, description and single call to action Pill Button.

The banner can be permanently fixed or dismissible, it also has a full width or column width state.

[Design Documentation](https://www.figma.com/file/eBY9GFogdepN8P5UWEclrE/One-Flow?node-id=1850%3A150583)

Components

Photo or Icon Badge
Title
Description
Pill Button
Cross - can be fixed OR dismissible
Background
Default
Light mode - Blue 100
Dark mode - Blue 800
Green - Shareville
Light mode - Green 100
Dark mode - Green 800
White
Light mode - White
Dark mode - Grey 800

### CODE REVIEW

**Review by kk-11 — APPROVED** [review]

**Review by jonnakollin — APPROVED** [review]

File: src/Molecules/PromotionBanner/PromotionBanner.types.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  /** @default true */
+  dismissible?: boolean;
+  /** @default null */
+  onClose?: () => void;
+  mobileBadgeContent?: React.ReactNode;
+  /** @default module */
+  scope?: 'module' | 'page';
+  title: string | React.ReactChild | React.ReactChild[];
+  /** @default false */
+  // isMobile?: boolean;
```
Comment (hardikkaji) [inline src/Molecules/PromotionBanner/PromotionBanner.types.ts:20]: remove this!

File: src/Molecules/PromotionBanner/PromotionBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    } else if (backgroundColor === 'white') {
+      bg = t.color.bubbleBackground;
+    } else if (typeof backgroundColor === 'function') {
+      bg = () => backgroundColor(theme);
+    }
+
+    return bg;
+  };
+
+  const handleClose = () => {
```
Comment (hardikkaji) [inline src/Molecules/PromotionBanner/PromotionBanner.tsx:65]: try function with useCallback!

File: src/Molecules/PromotionBanner/PromotionBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  dismissible,
+  mobileBadgeContent,
+  onClose,
+  scope = 'module',
+  title,
+}) => {
+  const isMobile = useMedia((t) => t.media.lessThan(t.breakpoints.sm));
+  const [showPromotion, setShowPromotion] = useState(true);
+
+  const getColor = (t: any) => {
```
Comment (hardikkaji) [inline src/Molecules/PromotionBanner/PromotionBanner.tsx:52]: try function with useCallback!

Comment (hardikkaji) [inline src/Molecules/PromotionBanner/PromotionBanner.tsx:52]: There is Theme type in theme.types.ts! I would recommend to use it!

**Review by hardikkaji — CHANGES_REQUESTED** [review]

File: src/Molecules/PromotionBanner/PromotionBanner.types.ts (hunk 1 above)
Comment (mic1000) [reply src/Molecules/PromotionBanner/PromotionBanner.types.ts:20]: I missed that.

**Pull request merged**